bundles/org.openhab.binding.flume/src/main/java/org/openhab/binding/flume/internal/handler/FlumeBridgeHandler.java

```diff
@@ -0,0 +1,303 @@
+
+        for (Thing t : things) {
+            if (!t.getThingTypeUID().equals(THING_TYPE_DEVICE)) {
+                continue;
+            }
+
+            FlumeDeviceHandler handler = (FlumeDeviceHandler) t.getHandler();
+            if (handler != null && handler.getId().equals(id)) {
+                return handler;
+            }
+        }
+
+        return null;
+    }
+
+    public void handleApiException(Exception e) {
+        if (e instanceof FlumeApiException) {
+            if (((FlumeApiException) e).isConfigurationIssue()) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                        e.getLocalizedMessage());
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                        e.getLocalizedMessage());
+            }
+        } else if (e instanceof IOException) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getLocalizedMessage());
+        } else if (e instanceof InterruptedIOException) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getLocalizedMessage());
+        } else if (e instanceof InterruptedException) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getLocalizedMessage());
+        } else if (e instanceof TimeoutException) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getLocalizedMessage());
+        } else if (e instanceof ExecutionException) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getLocalizedMessage());
+        } else {
+            // capture in log since this is an unexpected exception
+            logger.warn("Unhandled Exception", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.NONE, e.toString());
+        }
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        // cloud handler has no channels
+    }
+
+    private void pollDevices() {
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            // try to go online if it is offline due to communication error
+            if (getThing().getStatusInfo().getStatusDetail() == ThingStatusDetail.COMMUNICATION_ERROR) {
+                goOnline();
+            }
+            return;
+        }
+
+        // refresh listDevicesCache if necessary
+        if (apiListDevicesCache.isExpired()) {
+            refreshDevices(true);
+        }
+
+        List<Thing> things = getThing().getThings();
+
+        for (Thing t : things) {
+            if (!t.getThingTypeUID().equals(THING_TYPE_DEVICE)) {
+                continue;
+            }
+
+            FlumeDeviceHandler handler = (FlumeDeviceHandler) t.getHandler();
+            if (handler == null) {
```

Comment:
cast and null check in oneliner:
```suggestion
            if (t.getHandler() instanceof FlumeDeviceHandler) {
```